recipes/_deps_centos.rb

```diff
@@ -0,0 +1,112 @@
+#
+# Cookbook Name:: mcrouter
+# Recipe:: _build_deps.centos.rb
+#
+# Copyright 2015 EverTrue, Inc.
+# Copyright 2017 Roblox, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+include_recipe 'build-essential'
+include_recipe 'yum'
+include_recipe 'yum-epel'
+
+cache_dir = Chef::Config[:file_cache_path]
+
+%w(
```

Comment:
There's probably a lot of duplication in terms of dependencies, but this is probably the most robust way to handle it for the moment: just list each package out and let the package manager figure it out. Otherwise you're at the whim of distro maintainers changing dependencies on and breaking your cookbook.

Just my 2 cents.